test/pytest/test_example_stateful_sequence_continuous_batching_http.py

```diff
@@ -0,0 +1,363 @@
+                "1 4 -1 -1 -1",
+            ),
+        )
+        t1 = threading.Thread(
+            target=__infer_stateful,
+            args=(
+                model_name,
+                "seq_1",
+                "2 6 12 20 30",
+            ),
+        )
+
+        t0.start()
+        t1.start()
+
+        t0.join()
+        t1.join()
+    finally:
+        test_utils.unregister_model(model_name)
+
+        # Clean up files
+        shutil.rmtree(Path(model_store) / model_name)
+
+
+def __infer_stateful(model_name, sequence_id, expected):
+    start = True
+    prediction = []
+    for idx in range(5):
+        if sequence_id == "seq_0":
+            idx = 2 * idx
+        elif sequence_id == "seq_1":
+            idx = 2 * idx + 1
+        if start is True:
+            response = requests.post(
+                url=f"http://localhost:8080/predictions/{model_name}",
+                data=str(idx + 1).encode(),
+            )
+            s_id = response.headers.get("ts_request_sequence_id")
+            if sequence_id == "seq_0":
+                headers_seq_0 = {
+                    "ts_request_sequence_id": s_id,
+                }
+            elif sequence_id == "seq_1":
+                headers_seq_1 = {
+                    "ts_request_sequence_id": s_id,
+                }
+            start = False
+        else:
+            response = requests.post(
+                url=f"http://localhost:8080/predictions/{model_name}",
+                headers=headers_seq_0 if sequence_id == "seq_0" else headers_seq_1,
+                data=str(idx + 1).encode(),
+            )
+        prediction.append(response.text)
+
+    print(f"infer_stateful prediction={str(' '.join(prediction))}")
+    assert str(" ".join(prediction)) == expected
+
+
+def __infer_stateful_end(model_name, sequence_id, expected):
+    prediction = []
+    start = True
+    end = False
+    for idx in range(5):
+        if idx == 4:
+            end = True
+        if sequence_id == "seq_0":
+            idx = 2 * idx
+        elif sequence_id == "seq_1":
+            idx = 2 * idx + 1
+        if end is True:
+            idx = -1
+
+        if start is True:
+            response = requests.post(
+                url=f"http://localhost:8080/predictions/{model_name}",
+                data=str(idx + 1).encode(),
+            )
+            s_id = response.headers.get("ts_request_sequence_id")
+            if sequence_id == "seq_0":
+                headers_seq_0 = {
+                    "ts_request_sequence_id": s_id,
+                }
+            elif sequence_id == "seq_1":
+                headers_seq_1 = {
+                    "ts_request_sequence_id": s_id,
+                }
+            start = False
+        else:
+            response = requests.post(
+                url=f"http://localhost:8080/predictions/{model_name}",
+                headers=headers_seq_0 if sequence_id == "seq_0" else headers_seq_1,
+                data=str(idx + 1).encode(),
```

Comment:
For seq_0 we expect the last value to stay constant as we send 0, but we also send 0 to seq_1. Why do we expecty the last value to be 30 there?

Reply:
b/c seq_1 call infer_stateful, not infer_stateful_end

Reply:
got it, missed that, thanks